cassandra/cluster.py

```diff
@@ -1202,7 +1211,7 @@ def _make_connection_kwargs(self, address, kwargs_dict):
         if self._auth_provider_callable:
             kwargs_dict.setdefault('authenticator', self._auth_provider_callable(address))
 
-        kwargs_dict.setdefault('port', self.port)
+        kwargs_dict.setdefault('port', self.contact_points_resolved_w_port[address])
```

Comment:
This only works if you specify all the nodes in the contact list, otherwise when the new nodes are discovered it will give a KeyError

Reply:
That's a good point. 
What would be a good solution to the problem? New Cluster subclass?